opentelemetry-etw-logs/src/logs/exporter/options.rs

```diff
@@ -0,0 +1,169 @@
+            let result = callback(log_record);
+            if validate_etw_event_name(result).is_ok() {
+                return result;
+            }
+        }
+        self.default_event_name()
+    }
+
+    pub(crate) fn etw_event_name_from_callback(
+        mut self,
+        callback: impl Fn(&SdkLogRecord) -> &str + Send + Sync + 'static,
+    ) -> Self {
+        self.event_name_callback = Some(Box::new(callback));
+        self
+    }
+}
+
+trait EventNameCallback: Fn(&SdkLogRecord) -> &str + Send + Sync + 'static {}
+
+impl<F> EventNameCallback for F where F: Fn(&SdkLogRecord) -> &str + Send + Sync + 'static {}
+
+impl std::fmt::Debug for dyn EventNameCallback {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("ETW event name callback")
+    }
+}
+
+fn validate_etw_event_name(event_name: &str) -> Result<(), Box<dyn Error>> {
+    if event_name.is_empty() {
+        return Err("Event name cannot be empty.".into());
+    }
+    // TODO: Finish validation for ETW event name
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::super::common::test_utils;
+    use super::*;
+
+    #[test]
+    fn test_get_event_name() {
+        use opentelemetry::logs::LogRecord;
+
+        let mut log_record = test_utils::new_sdk_log_record();
+
+        let options = test_utils::test_options();
+
+        let result = options.get_etw_event_name(&log_record);
+        assert_eq!(result, "Log");
+
+        log_record.set_event_name("event-name");
+        let result = options.get_etw_event_name(&log_record);
+        assert_eq!(result, "Log");
+
+        log_record.set_target("target-name");
+        let result = options.get_etw_event_name(&log_record);
+        assert_eq!(result, "Log");
+    }
+
+    #[test]
+    fn test_get_event_name_from_callback_returning_const_value() {
+        use opentelemetry::logs::LogRecord;
+
+        let mut log_record = test_utils::new_sdk_log_record();
+
+        let options =
+            Options::new("test-provider-name").etw_event_name_from_callback(|_| "CustomEvent");
+
+        let result = options.get_etw_event_name(&log_record);
+        assert_eq!(result, "CustomEvent");
+
+        log_record.set_event_name("event-name");
+        let result = options.get_etw_event_name(&log_record);
+        assert_eq!(result, "CustomEvent");
+
+        log_record.set_target("target-name");
+        let result = options.get_etw_event_name(&log_record);
+        assert_eq!(result, "CustomEvent");
+    }
+
+    #[test]
+    fn test_get_event_name_from_callback_returning_name() {
+        use opentelemetry::logs::LogRecord;
+
+        let mut log_record = test_utils::new_sdk_log_record();
+
+        let options = Options::new("test-provider-name")
+            .etw_event_name_from_callback(|log_record| log_record.event_name().unwrap_or_default());
+
+        let result = options.get_etw_event_name(&log_record);
+        assert_eq!(result, "Log");
```

Comment:
interesting that "Log" is still the name, though the callback is always returning something... Is this trying to test that `invalid_event_name_from_callback_will_use_default` ?

Reply:
Got it. It's better to have a targeted test for the same. `invalid_event_name_from_callback_will_use_default` would be my suggestion.

Not a blocker.